### PR TITLE
Improve generate-api-diff output to better see errors

### PR DESCRIPTION
**Pull request by JamesBurnside**
# What

- Add `-v` to the rush build commands so errors are not swallowed
   - image
- Add a failed summary which prints the executed commands that a developer can run locally themself
   - image
- Fix cleanup to restore build flavor in cleanup
- Remove extra redundant feature.js destage, instead just build baseline first

# Why

When it failed important errors were being swallowed and only confusing errors were left.

# How Tested

Ran a bunch locally

### CODE REVIEW

File: common/scripts/generate-api-diff.mjs [hunk 1]
```diff
@@ -60,21 +77,29 @@ async function setup() {
...
+  // Backup the files that will be modified by the build process
   backupFiles(filesToBackup);
+
+  // Store the active development flavor to restore to once the cmd finishes
+  initialDevelopmentFlavor = getBuildFlavor();
 }
 
 /** To be called at the end of the script to restore the files that were modified by the build process */
 async function cleanup() {
+  console.log('Running cleanup');
```
Comment (emlynmac) [inline common/scripts/generate-api-diff.mjs:89]: Nit: should we remove this log?

**Review by emlynmac — APPROVED** [review]

File: common/scripts/stage-feature.mjs [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+  const args = yargs(argv.slice(2))
+    .usage('$0 [options]', 'Use this script to generate api.json files for beta features for uploading to apiview.dev for review.')
+    .example([
+      ['$0 --feature my-feature --option alphaToBeta', 'Mark feature as alpha from beta.']
+    ])
+    .options({
+      feature: {
+        alias: 'f',
+        type: 'string',
+        describe: 'Feature name to generate api files for. This should match the name in the babelrc file'
```
Comment (edwardlee-msft) [inline common/scripts/stage-feature.mjs:72]: nit: technically the babelrc file with the list of feature names was moved from babelrc.js to feature.js in the babel folder.

Comment (JamesBurnside) [reply common/scripts/stage-feature.mjs:72]: You are very correct, will update this

File: common/scripts/generate-api-diff.mjs [hunk 3]
```diff
@@ -34,22 +34,39 @@ const filesToBackup = [
...
-  if (!wasSuccessful) {
-    exit(-1);
+  // Ouput the result of the process _after_ the cleanup process has run to ensure the logs are not lost
+  if (result) {
+    console.log(`\nDone generating api.json files, files can be found at:`);
+    console.log(`  - ${result.baselineFilePath}`);
+    console.log(`  - ${result.featureFilePath}`);
+    console.log(`Please upload these files to https://apiview.dev/ for review.`);
+  } else {
+    console.error(`\nScript failed to generate api.json files.\nInspect the logs above for more information on errors.\nAlternatively, this tool ran the following commands you can run locally to reproduce the error: \n${commandsRun.map(c => `  - ${c}`).join('\n')}`);
```
Comment (edwardlee-msft) [inline common/scripts/generate-api-diff.mjs:68]:
```suggestion
    console.error(`\nScript failed to generate api.json files.\nInspect the logs above for more information on errors.\nAlternatively, this tool runs the following commands that you can run locally to reproduce the error: \n${commandsRun.map(c => `  - ${c}`).join('\n')}`);
```

**Review by edwardlee-msft — APPROVED** [review]

**Pull request merged**